File: backend/src/core/authorizer.py
from ..core.state import Store
from . import account, transaction


def process_operation(store: Store, operation: dict) -> tuple[dict, list[str]]:
    if "account" in operation:
        account_dict, violations = account.process_account(store, operation["account"])
        result_account = {}
        if account_dict:
            result_account = {
                "active-card": account_dict.active_card,
                "available-limit": account_dict.available_limit
            }
        return result_account, violations
    
    if "transaction" in operation:
        return transaction.process_transaction(store, operation["transaction"])
    
    return {}, []

File: frontend/src/core/authorizer.js
import * as account from './account.js';
import * as transaction from './transaction.js';

export function processOperation(store, operation) {
  if (operation.account) {
    const [accountResult, violations] = account.processAccount(store, operation.account);
    const accountDict = accountResult
      ? { 'active-card': accountResult.activeCard, 'available-limit': accountResult.availableLimit }
      : {};
    return [accountDict, violations];
  }

  if (operation.transaction) {
    return transaction.processTransaction(store, operation.transaction);
  }

  return [{}, []];
}